postgres/alter_system.py

```diff
@@ -59,55 +58,44 @@ def human_to_int(value):
     return kB * int(value.strip("kB"))
   return int(value)
 
-# Read from /sys file system
-def get_cgroup_resources(file):
-  with open(file, 'r') as proc_fil:
-      # Read cmdline value.
-      data = proc_fil.read()
-      # Make it printable.
-      ret_val = data.rstrip('\n')
-  return ret_val
-
 # Get memory limit
+# returns amount of memory in bytes as int
 def get_mem():
   # get memory numbers from host
-  sysmem = psutil.virtual_memory()
-  # first try cgroup v2 file
-  file = "/sys/fs/cgroup/memory.max"
-  if os.path.isfile(file):
-      mem = get_cgroup_resources(file)
-      if mem == 'max':
-        return sysmem.total
-      return int(mem)
-  # then try cgroup v1 file
-  file = "/sys/fs/cgroup/memory/memory.limit_in_bytes"
-  if os.path.isfile(file):
-      mem = int(get_cgroup_resources(file))
-      if mem > sysmem.total:
-        return sysmem.total
-      return mem
-  # default return max memory
-  return sysmem.total
+  sysmem = psutil.virtual_memory().total
+  # cgroup v2
+  try:
+      mem = open("/sys/fs/cgroup/memory.max").read().rstrip()
+  except IOError:
+      # cgroup v1
+      try:
+          mem = open("/sys/fs/cgroup/memory/memory.limit_in_bytes").read().rstrip()
+      # no cgroup memory limits configured, assuming max
+      except IOError:
+          mem = "max"
+
+  # max in cgroup v2, -1 in v1
+  # if max or cgroup limit mem bigger than sysmem, use sysmem, else use cgroup mem limit
+  return sysmem if mem == "max" or int(mem) > sysmem else int(mem)
```

Comment:
Ups a bit of a copy paste error, I'm a bit confused what the fallback for `/sys/fs/cgroup/memory/memory.limit_in_byte` is, if it is `-1` this is of course incorrect.

Reply:
I am not sure, if you resolved your confusion yourself. In cgroups v1 you get -1 if there is no memory limit, in v2 it's "max". It looks to me, that result -1 (no limit in v1) is currently not covered. So why not use the same pattern in line 79 as you did in line 98:

`return sysmem if mem in ("max", -1) or int(mem) > sysmem else int(mem)`

Reply:
Cool! Happy this change is appreciated.

I basically did the same as it was in your code.

https://github.com/hzi-braunschweig/SORMAS-Docker/pull/221/commits/f8648c526f60933f303fdc8292008629ed5fe254#diff-d55335daa9b1401ee6fd0c285186934876f5c891a9b157f8e3e28f4ba28d78a1R83-R88

I guess good that I had a look, I'll adjust the commit accordingly.

Reply:
Ah found another issue it should be `'-1'` not `-1` of course. I've updated the script